### PR TITLE
Simplify RegistryEndpointCaller code

### PR DESCRIPTION
**Pull request by chanseokoh**
I notice that `RequestState.authorization` is always the value given to the `RegistryEndpointCaller` constructor, so there is not much reason to carry this around in `RequestState`.

After that, `RequestState` becomes a meaningless class holding only a single `URL` field, so it seems obvious to get rid of it.

### CODE REVIEW

File: jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ public RegistryAuthenticator initialize()
         throw new RegistryAuthenticationFailedException(ex);
 
       } catch (InsecureRegistryException ex) {
-        // HTTP is not allowed, so just return null.
+        // Cannot skip certificate validation or use HTTP, so just return null.
```
Comment (briandealwis) [inline jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java]: This doesn't seem related.

Comment (TadCordle) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java]: I think it's small enough that it's fine to keep it in the PR

Comment (chanseokoh) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java]: No worries. Will have this in a later PR.
